src/api/java/com/minecolonies/api/compatibility/CompatibilityManager.java

```diff
@@ -219,9 +225,35 @@ public void discover(final boolean serverSide)
      */
     private void discoverBlockList()
     {
-        allBlocks = ImmutableList.copyOf(StreamSupport.stream(Spliterators.spliteratorUnknownSize(ForgeRegistries.ITEMS.iterator(), Spliterator.ORDERED), false)
+        final List<ItemStack> stacks = StreamSupport.stream(Spliterators.spliteratorUnknownSize(ForgeRegistries.ITEMS.iterator(), Spliterator.ORDERED), false)
+                                           .filter(item -> !(item instanceof EnchantedBookItem || item instanceof PotionItem))
                                            .map(ItemStack::new)
-                                           .collect(Collectors.toList()));
+                                           .collect(Collectors.toList());
+        for(Enchantment enchantment : Registry.ENCHANTMENT)
+        {
+            for(int i = enchantment.getMinLevel(); i <= enchantment.getMaxLevel(); ++i)
+            {
+                stacks.add(getEnchantedItemStack(new EnchantmentData(enchantment, i)));
+            }
+        }
+
+        for(Potion potion : Registry.POTION)
+        {
+            if (potion != Potions.EMPTY)
+            {
+                stacks.add(PotionUtils.addPotionToItemStack(new ItemStack(Items.POTION), potion));
+            }
+        }
+
+        for(Potion potion : Registry.POTION)
+        {
+            if (!potion.getEffects().isEmpty())
+            {
+                stacks.add(PotionUtils.addPotionToItemStack(new ItemStack(Items.TIPPED_ARROW), potion));
+            }
+        }
+
+        allBlocks = ImmutableList.copyOf(stacks);
     }
```

Comment:
This method is either named wrong, or does literally not what is written on the can.
Rename the method.

Also make this a parallel stream. With more then 1000+ entries in that list of items this should help.